src/containers/Layout/NotificationDrawerWrapper.tsx

```diff
@@ -1,42 +1,72 @@
 import {
+  Button,
   NotificationDrawer,
   NotificationDrawerBody,
   NotificationDrawerHeader,
   NotificationDrawerList,
   NotificationDrawerListItem,
   NotificationDrawerListItemBody,
-  NotificationDrawerListItemHeader
+  NotificationDrawerListItemHeader,
+  ButtonVariant
 } from '@patternfly/react-core';
-import React, { useContext } from 'react';
+import React, { useContext, useEffect, useState } from 'react';
 import { AppContext } from '../../context/context';
+import { TimesIcon, TimesCircleIcon } from "@patternfly/react-icons";
+import { NotificationActionTypes } from "../../context/actions/types";
+import { NotificationItem } from "../../context/reducers/notificationReducer";
 
 interface NotificationDrawerWrapperProps {
   onClose: () => void;
 }
 
 const NotificationDrawerWrapper: React.FC<NotificationDrawerWrapperProps> = ({ onClose }) => {
-  const { state: { notifications } } = useContext(AppContext);
+  const [disabled, setDisabled] = useState(true);
+  const { state: { notifications }, dispatch } = useContext(AppContext);
+
+  const onNotificationClose = (index: number) => {
+    let newNotifications: NotificationItem[] = notifications.slice(0);
+    newNotifications.splice(index, 1);
+
+    dispatch({
+      type: NotificationActionTypes.REMOVE,
+      payload: {
+        notifications: newNotifications
+      }
+    });
+  }
+
+  useEffect(() => {
+    setDisabled(notifications.length === 0)
+  }, [notifications.length]);
 
   return (
     <NotificationDrawer>
-      <NotificationDrawerHeader count={notifications.length} onClose={() => onClose()}>
+      <NotificationDrawerHeader count={notifications.length}>
+        <Button variant="tertiary" aria-label="Clear All" isDisabled={disabled} onClick={() => {
+          dispatch({
+            type: NotificationActionTypes.CLEAR
+          });
+        }}>Clear All</Button>
```

Comment:
I would suggest extracting the `onClick` logic into a separate function instead of keeping it inside HTML template. One of the reasons is because it helps with debugging the codebase when separation of concerns is applied. As an example, if this was in a separate function, one would be able to put breakpoints within the JS file.

Reply:
Great suggestion and coding practice that I overlooked!